homeassistant/components/panasonic_viera/media_player.py

```diff
@@ -168,79 +168,136 @@ def supported_features(self):
             return SUPPORT_VIERATV | SUPPORT_TURN_ON
         return SUPPORT_VIERATV
 
+    @property
+    def device_info(self):
+        """Return device specific attributes."""
+        return {
+            "name": self.name,
+            "identifiers": {(DOMAIN, self.unique_id)},
+            "manufacturer": DEVICE_MANUFACTURER,
+        }
+
+    def update(self):
+        """Retrieve the latest data and renew connection."""
+        if self._remote:
+            try:
+                self._muted = self._remote.get_mute()
+                self._volume = self._remote.get_volume() / 100
+                self._state = STATE_ON
+            except OSError:
+                self._state = STATE_OFF
+
+        old_remote = self._remote
+        try:
+            if self._app_id and self._encryption_key:
+                self._remote = RemoteControl(
+                    self._host,
+                    self._port,
+                    app_id=self._app_id,
+                    encryption_key=self._encryption_key,
+                )
+            else:
+                self._remote = RemoteControl(self._host, self._port)
+        except Exception as err:
+            if old_remote:
+                _LOGGER.error("Could not establish remote connection: " + repr(err))
+
+    def send_key(self, key):
+        """Send a key to the tv and handles exceptions."""
+        if self._remote:
+            try:
+                self._remote.send_key(key)
+                self._state = STATE_ON
+            except OSError:
+                self._state = STATE_OFF
+                return False
+            return True
+
     def turn_on(self):
         """Turn on the media player."""
         if self._mac:
             self._wol.send_magic_packet(self._mac, ip_address=self._broadcast)
```

Comment:
Please remove the WOL feature and replace this with a general script config option. See the samsung mediaplayer for example.

https://github.com/home-assistant/core/blob/2d1002d40da81f5540e88022e3ac8039f39cd163/homeassistant/components/samsungtv/__init__.py#L32

https://github.com/home-assistant/core/blob/2d1002d40da81f5540e88022e3ac8039f39cd163/homeassistant/components/samsungtv/media_player.py#L245-L248

Reply:
How would this be configurable in a config flow?

Reply:
Or should it be something exclusive for YAML?

Reply:
It wouldn't currently be configurable in the config flow. We don't want options in the main config flow. Only required items to establish the connection. It's ok to require config yaml for this option.

Reply:
That makes things much easier. Then I'll keep only `host` and `name` in the config flow, just like in the Samsung TV integration